src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1186,9 +1187,9 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
                     continue;
                 }
 
-                if (okToDowngradeToNeither)
+                if (!considerRefKinds || Conversions.ClassifyImplicitConversion(type1, type2, ref useSiteDiagnostics).Kind != ConversionKind.Identity)
                 {
-                    Debug.Assert(Conversions.ClassifyImplicitConversion(type1, type2, ref useSiteDiagnostics).Kind != ConversionKind.Identity);
+                    Debug.Assert(considerRefKinds || Conversions.ClassifyImplicitConversion(type1, type2, ref useSiteDiagnostics).Kind != ConversionKind.Identity);
```

Comment:
I don't believe the condition is the same. If considerRefKinds is false, conversion between parameter types isn't classified by the if condition. This assert is here to verify the assumption that the conversion is never an identity in that case and we can skip classification as an optimization.